### ISSUE TITLE
Unable to specify AWS s3 region

### ISSUE THREAD

**Issue by erichoracek** [blender/Rome#4]
When attempting to try out the executable for fixing #2, I ran into the issue where attempting to upload binaries to AWS is failing. This is due to our carthage cache bucket being hosted in the `us-west-2` region.

The `upload` command fails with the following error:

```
Error uploading FrameworkName : The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.
```

I have the region specified in my `.aws/config` file, but this doesn't seem to be having any effect either.

It would appear that the issue is `NorthVirginia` in this file https://github.com/blender/Rome/blob/master/app/Main.hs#L23

**Comment by blender** [comment]
Hi, thanks for reporting. I'll find a way to a either avoid specifying a region all together or the opposite

**Comment by blender** [comment]
@erichoracek can you access your file manually using as base S3 url ``s3.amazonaws.com``or do you only success if you use ``s3-us-west-2.amazonaws.com`` ?

**Comment by erichoracek** [comment]
The region is required, e.g. when using the `aws` CLI

```
$ aws s3 ls s3://bucket.name                    

A client error (PermanentRedirect) occurred when calling the ListObjects operation: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint: s3.amazonaws.com
You can fix this issue by explicitly providing the correct region location using the --region argument, the AWS_DEFAULT_REGION environment variable, or the region variable in the AWS CLI configuration file.  You can get the bucket's location by running "aws s3api get-bucket-location --bucket BUCKET".
```

**Comment by erichoracek** [comment]
To answer your question: I don't think it's required when you're accessing an S3 URL—it only seems to be required when interacting with the APIs.

**Comment by blender** [comment]
I spoke with the maintainer for the AWS library I'm using and I was told that a region is request for v4 request signing. I will implement region discovery from ``.aws/config``

**Comment by blender** [comment]
@erichoracek Please try fix in pre-release https://github.com/blender/Rome/releases/tag/v0.3.0.2

**Comment by erichoracek** [comment]
This is fixed in 3.0.2—thanks!
